lib/ui/pointer.dart

```diff
@@ -380,6 +384,29 @@ class PointerData {
   /// The current angle of the pan/zoom in radians, with 0.0 as the initial angle.
   final double rotation;
 
+  // An optional function that allows the framework to respond to the event
+  // that triggered this PointerData instance.
+  final PointerDataRespondCallback? _onRespond;
```

Comment:
We *can* inline a function type here without creating a one-off `typedef`. Is this a guideline or something?

Reply:
> We can inline a function type here without creating a one-off `typedef`. Is this a guideline or something?

@kevmoo I went back and forth on this (I originally had this as an inline function type). The [Style guide for Flutter repo](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#naming-rules-for-typedefs-and-function-variables) that @goderbauer linked says this:

> Prefer using `typedef`s to declare callbacks. Typedefs benefit from having documentation on the type itself and make it easier to read and find common callsites for the signature.

I don't have strong feelings about it (but when we were bikeshedding in parameter name, it was a chore to rename it in several, unconnected parts, and that's when I ended up adding the `typedef`).